app/models/course/registration.rb

```diff
@@ -0,0 +1,54 @@
+class Course::Registration < ActiveRecord::Base
+  stampable
+
+  scope :staff, -> { where(role: [roles[:teaching_assistant], roles[:manager], roles[:owner]]) }
+
+  enum role: CourseUser.roles
+
+  belongs_to :course
+  belongs_to :user
+
+  # Approve the registration record by creating a CourseUser of the user to the course with the
+  # requested role.
+  def approve!
+    course.course_users.create!(user_id: user_id, role: role)
+  end
+
+  # Process the registration records of given IDs and then delete them.
+  #
+  # @param [Course] the course that the registration records belong to.
+  # @param [Array<Int>] the IDs of registration records
+  # @options approve: If specified a true equivalent value, the Registration#approve! method will
+  #                   be called. Default to false.
+  #          process_all: if specified a true equivalent value, all registration to the course
+  #                       will be processed. Default to false.
+  #          role: If specified a value in the Registration.roles enum, only registrations of that
+  #                role will be processed. Other IDs are ignored. Default to nil.
+  def self.process!(course, registration_ids, options = {})
+    if options[:process_all]
+      registrations = course.registrations
+    else
+      registration_ids ||= []
+      registrations = registration_ids.uniq.
+        map { |id| Course::Registration.find_by_id_and_course_id(id, course.id) }.
+        select { |registration| registration }
+    end
+
+    if options[:role]
+      registrations = registrations.select { |registration| registration.send("#{options[:role]}?") }
+    end
+
+    processed_ids = []
+    registrations.each do |registration|
+      if options[:approve]
```

Comment:
Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.